source/renderer/app/stores/NetworkStatusStore.js

```diff
@@ -488,6 +492,12 @@ export default class NetworkStatusStore extends Store {
             (isLocalBlockHeightSyncing || isNetworkBlockHeightSyncing);
         });
 
+        if (this.isNodeRestarting && this.isNodeSyncing) {
```

Comment:
@daniloprates can you please clarify this part? 
Wouldn't it be better to shift this logic which sets `this.isNodeRestarting = false;` under `_handleCardanoNodeStateChange` method?

Reply:
@daniloprates I still think this is worth improving 🙏

Reply:
The reason it needs to be there is that, after Node is Running, there's a gap before `hasStartedReceivingBlocks` is `true`, which is when the flickering happens.

The way it was done, not only awaits for Node to be Running, but also for the first block to be received.

Reply:
@daniloprates are you sure? `isConnected` requires for the `isNodeSyncing` to be `true` which implies that `hasStartedReceivingBlocks` is `true`?

Reply:
Investigating

Reply:
Found the issue. `_requestCardanoStatus` was overriding the current status, wrongly setting `isNodeSyncing` to `true`. Simplified the code, now.

Reply:
Tanks @daniloprates - will review this first thing tomorrow morning!